src/sources/kafka.rs

```diff
@@ -439,15 +509,66 @@ impl ReceivedMessage {
         }
     }
 
-    fn apply(&self, keys: &Keys<'_>, event: &mut Event) {
+    fn apply(&self, keys: &Keys<'_>, event: &mut Event, log_namespace: LogNamespace) {
         if let Event::Log(ref mut log) = event {
-            log.insert(keys.source_type, Bytes::from("kafka"));
-            log.insert(keys.timestamp, self.timestamp);
-            log.insert(keys.key_field, self.key.clone());
-            log.insert(keys.topic, Value::from(self.topic.clone()));
-            log.insert(keys.partition, Value::from(self.partition));
-            log.insert(keys.offset, Value::from(self.offset));
-            log.insert(keys.headers, Value::from(self.headers.clone()));
+            if let LogNamespace::Vector = log_namespace {
+                // We only want an auto-generated timestamp in the Vector namespace.
+                log_namespace.insert_standard_vector_source_metadata(
+                    log,
+                    KafkaSourceConfig::NAME,
+                    Utc::now(),
+                );
+            } else {
+                log.insert("source_type", KafkaSourceConfig::NAME);
+
+                log_namespace.insert_source_metadata(
+                    KafkaSourceConfig::NAME,
+                    log,
+                    Some(LegacyKey::Overwrite(keys.key_field)),
+                    path!(default_key_field().as_str()),
```

Comment:
I'd rather us hardcode the value here - rather than relying on the default function there, it seems less likely to accidentally change.

Reply:
Ditto for the other keys, but this is just a feelz thing - eventually the default value won't exist because it'll be a `const` value.